src/Security/UserClassBuilder.php

```diff
@@ -22,21 +25,39 @@
  */
 final class UserClassBuilder
 {
-    public function addUserInterfaceImplementation(ClassSourceManipulator $manipulator, UserClassConfiguration $userClassConfig)
+    public function addUserInterfaceImplementation(ClassSourceManipulator $manipulator, UserClassConfiguration $userClassConfig): void
     {
         $manipulator->addInterface(UserInterface::class);
 
         $this->addGetUsername($manipulator, $userClassConfig);
 
         $this->addGetRoles($manipulator, $userClassConfig);
 
-        $this->addGetPassword($manipulator, $userClassConfig);
-
-        $this->addGetSalt($manipulator, $userClassConfig);
+        if (06000 < Kernel::VERSION_ID) {
+            $this->addGetPassword($manipulator, $userClassConfig);
+            $this->addGetSalt($manipulator, $userClassConfig);
+        }
```

Comment:
I understand why it's written this way, but I'm having a hard time following the flow.

Could we make `addUserPasswordInterface()` private and call it from `addUserInterfaceImplementation()` (we could rename both methods if we want to for clarity). Then `addUserPasswordInterface()` might be called `addPasswordImplementation()` and it would have all the logic for adding the interface (if needed) and generating the fields. And we wouldn't need to check if the `getPassword()` method already exists.

Reply:
done